### PR TITLE
<*Fiel /> components → Adjust childs spacing 😐 📐

### PR DESCRIPTION
**Pull request by lluia**
### help text and error message spacing

The _white-space_ after the **label** on `<TextField />` was off. I suspect it comes from [an update we did in `<InputLabel />`](https://github.com/zopaUK/react-components/pull/170) a few weeks ago.

I made the spacing consistent as well in `<DropdownField />` and `<DropdownFiltered />`.

### default margin on radio and checkboxes

There was a default `margin-bottom` applied without need to `<RadioField />` and `<CheckboxField />`. I removed it and made it appear only when more than one element is rendered through `:last-of-type`. 💭

## Notes  &nbsp;👀

This might justify picking up again [the spike on visual diff](https://github.com/zopaUK/react-components/issues/14) with [Percy](percy.io/).

### CODE REVIEW

**Comment by poteirard** [comment]
Thanks a lot for this 🙏

File: src/components/molecules/CheckboxField/CheckboxField.tsx [hunk 1]
```diff
@@ -82,18 +83,24 @@ const Label = styled(InputLabel)`
   }
 `;
 
-const CheckboxField: React.FC<IField> = props => {
+const FieldError = styled(ErrorMessage)`
+  margin-top: 5px;
+`;
+
+const CheckboxField: React.FC<Omit<IField, 'helpText'>> = props => {
```
Comment (Blimeys) [inline src/components/molecules/CheckboxField/CheckboxField.tsx]: Do you want to accept a children here ? If not, you dont need FC

File: src/components/molecules/DropdownField/DropdownField.tsx [hunk 2]
```diff
@@ -15,23 +15,33 @@ export interface IDropdownFieldProps extends IField<HTMLSelectElement> {
   name: string;
 }
 
+export interface ILabelProps {
+  withHelpText?: boolean;
```
Comment (Blimeys) [inline src/components/molecules/DropdownField/DropdownField.tsx]: Might be good to assign a default value to `withHelpText` so its always a boolean and not a boolean or undefined

File: src/components/molecules/DropdownField/DropdownField.tsx [hunk 3]
```diff
@@ -15,23 +15,33 @@ export interface IDropdownFieldProps extends IField<HTMLSelectElement> {
...
+}
+
+const Label = styled(InputLabel)<ILabelProps>`
+  ${({ withHelpText }) => withHelpText && `margin-bottom: 0;`}
```
Comment (Blimeys) [inline src/components/molecules/DropdownField/DropdownField.tsx]: `withHelpText` or `withLabel` ? Im confused, maybe rename `HelpText` ?

File: src/components/molecules/TextField/TextField.tsx [hunk 4]
```diff
@@ -17,12 +17,23 @@ export interface IPrefixProps {
   prefix: string;
 }
 
-const TextFieldError = styled(ErrorMessage)`
+export interface ILabelProps {
+  withHelpText?: boolean;
+}
+
+const Label = styled(InputLabel)<ILabelProps>`
```
Comment (Blimeys) [inline src/components/molecules/TextField/TextField.tsx]: Same as above

File: src/components/molecules/DropdownField/DropdownField.tsx (hunk 3 above)
Comment (lluia) [reply src/components/molecules/DropdownField/DropdownField.tsx]: I can rename `<Help />` to `<HelpText />` 👌🏻

File: src/components/molecules/DropdownField/DropdownField.tsx (hunk 2 above)
Comment (lluia) [reply src/components/molecules/DropdownField/DropdownField.tsx]: Hmm I don't see the benefit of it vs the current implementation 🤔

Comment (Blimeys) [reply src/components/molecules/DropdownField/DropdownField.tsx]: So you narrow down the type. It strictly becomes a boolean.

**Pull request merged**